### DISCUSSION TITLE
Do You Publish Your Lua Modules?

### DISCUSSION THREAD

**Discussion by frankgreco** [Kong/kong#12719] [Help] [answered]
I want to use [kong](https://github.com/Kong/kong/tree/master)/[kong](https://github.com/Kong/kong/tree/master/kong)/[plugins](https://github.com/Kong/kong/tree/master/kong/plugins)/[jwt](https://github.com/Kong/kong/tree/master/kong/plugins/jwt) in a project since this is the only Lua+JWT module i've found that supports ES256. Do you publish this anywhere (i.e. luarocks) so that I can import it?

**Comment by Tieske** [comment, marked as answer]
We do not. But you could create a separate project with only a `rockspec` file that references the Kong git repo. That should just work. Make sure to point to a specific git-tag to ensure the contents is stable.

Here's an example of doing the same thing; [this repo](https://luarocks.org/modules/moteus/ltn12) publishes the `ltn12` module which is part of LuaSocket.

**Reply by frankgreco** [reply]
Thanks! This is the file that worked for me 👇🏼 
```
package = "kong"
version = "3.6.1-0"
rockspec_format = "3.0"
source = {
  url = "git+https://github.com/Kong/kong.git",
  tag = "3.6.1"
}
description = {
  summary = "Kong"
}
dependencies = {
  "lua >= 5.1"
}
build = {
  type = "builtin",
  modules = {
    ["kong.jwt"] = "kong/plugins/jwt/jwt_parser.lua"
  },
  install = {
    lua = {
      ["kong.jwt"] = "kong/plugins/jwt/jwt_parser.lua"
    }
  }
}

```

I can then do `require("kong.jwt")`

**Reply by Tieske** [reply]
some comments:

- You're probably better of picking a different name. The package is now called "kong" which collides with the existing Kong rockspecs. Maybe "kong-jwt-parser" or something?
- afaik you do not need the `build.install` object in the rockspec
- the rockspec revision (the `-0` in the version number) should start at 1, not 0 (will work, but wasn't intended like that)
